sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/implementation/OkHttpFluxRequestBody.java

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.okhttp.implementation;
+
+import com.azure.core.http.HttpHeaders;
+import com.azure.core.implementation.util.BinaryDataContent;
+import com.azure.core.util.logging.ClientLogger;
+import okhttp3.MediaType;
+import okio.BufferedSink;
+import reactor.core.Exceptions;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.time.Duration;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * An {@link okhttp3.RequestBody} subtype that sends {@link BinaryDataContent}
+ * as {@link Flux} of {@link ByteBuffer} in an unbuffered manner.
+ * This class accepts any {@link BinaryDataContent} as catch-all for backwards compatibility
+ * but ideally should be used only with reactive payloads.
+ */
+public class OkHttpFluxRequestBody extends OkHttpStreamableRequestBody<BinaryDataContent> {
+
+    private static final ClientLogger LOGGER = new ClientLogger(OkHttpFluxRequestBody.class);
+
+    private final AtomicBoolean bodySent = new AtomicBoolean(false);
+    private final Duration writeTimeout;
```

Comment:
Should `writeTimeout` also be applied to the `InputStream`-based request

Reply:
I'm not 100% sure we need this timeout. I.e. previous version was buffering the flux without such thing. But on the other hand it didn't block the flux in some arbitrary place (i.e. timeout would be applied in the blob client for instance).

I added this in case the source stalls so that we don't block OkHttp thread forever. I'm not worried about the network write here. InputStream doesn't have InputStream.read that would take timeout.